src/components/Pantry/Pantry.jsx

```diff
@@ -109,7 +110,7 @@ export default function Pantry(props) {
     let itemName = productForm.name.toLowerCase()
     let itemIndex = userProducts.findIndex(item => item.name === itemName)
 
-    let newProducts = [...userProducts]
+    let newProducts = userProducts.map(i => ({ ...i }))
```

Comment:
What's the difference between `[...userProducts]` and `userProducts.map(i=>({...i}))`?

Reply:
I changed it to this since [...userProducts] gets a shallow copy of the array. Since the array holds objects, I noticed that the objects themselves were not being copied. Mapping and copying each objects makes sure I get a deep copy.